test/calibration/test_calibrations.py

```diff
@@ -0,0 +1,801 @@
+        """Test that schedule replacement works as expected."""
+
+        self.cals.add_parameter_value(ParameterValue(0.25, self.date_time), "amp", (3,), "xp")
+        self.cals.add_parameter_value(ParameterValue(40, self.date_time), "σ", schedule="xp")
+        self.cals.add_parameter_value(ParameterValue(10, self.date_time), "β", (3,), "xp")
+
+        # Let's replace the schedule for qubit 3 with a double Drag pulse.
+        with pulse.build(name="xp") as sched:
+            pulse.play(Drag(160, self.amp_xp / 2, self.sigma, self.beta), self.drive)
+            pulse.play(Drag(160, self.amp_xp / 2, self.sigma, self.beta), self.drive)
+
+        expected = self.cals.parameters
+
+        # Adding this new schedule should not change the parameter mapping
+        self.cals.add_schedule(sched, (3,))
+
+        self.assertEqual(self.cals.parameters, expected)
+
+        # For completeness we check that schedule that comes out.
+        sched_cal = self.cals.get_schedule("xp", (3,))
+
+        self.assertTrue(isinstance(sched_cal.instructions[0][1].pulse, Drag))
+        self.assertTrue(isinstance(sched_cal.instructions[1][1].pulse, Drag))
+        self.assertEqual(sched_cal.instructions[0][1].pulse.amp, 0.125)
+        self.assertEqual(sched_cal.instructions[1][1].pulse.amp, 0.125)
+
+        # Let's replace the schedule for qubit 3 with a Gaussian pulse.
+        # This should change the parameter mapping
+        with pulse.build(name="xp") as sched2:
+            pulse.play(Gaussian(160, self.amp_xp / 2, self.sigma), self.drive)
+
+        # Check that beta is in the mapping
+        self.assertEqual(
+            self.cals.parameters[self.beta],
+            {ParameterKey("β", (3,), "xp")},
+        )
+
+        self.cals.add_schedule(sched2, (3,))
+
+        # Check that beta no longer maps to a schedule
+        self.assertEqual(self.cals.parameters[self.beta], set())
+
+    def test_parameter_filtering(self):
+        """Test that we can properly filter parameter values."""
+
+        self._add_parameters()
+
+        # Check that these values are split between the qubits.
+        amp_values = self.cals.parameters_table(parameters=["amp"], qubit_list=[(0,)])
+        self.assertEqual(len(amp_values), 1)
+
+        # Check that we have one value for sigma.
+        sigma_values = self.cals.parameters_table(parameters=["σ"])
+        self.assertEqual(len(sigma_values), 1)
+
+        # Check that we have two values for amp.
+        amp_values = self.cals.parameters_table(parameters=["amp"])
+        self.assertEqual(len(amp_values), 2)
+
+        amp_values = self.cals.parameters_table(parameters=["amp"], qubit_list=[(3,)])
+        self.assertEqual(len(amp_values), 1)
+
+        # Check to see if we get back the two qubits when explicitly specifying them.
+        amp_values = self.cals.parameters_table(parameters=["amp"], qubit_list=[(3,), (0,)])
+        self.assertEqual(len(amp_values), 2)
+
+
+class TestMeasurements(QiskitTestCase):
```

Comment:
`Acquire` is not managed by this manager? I think measurement _instruction_ should be measure + acquire. The acquire may have delay from the measure pulse to compensate the round-trip time (depends on controller). This delay seems to be a parameter what we want to calibrate.

Reply:
Good point, I added a test for this case and added the needed channels. See here: 3c56906.